filebeat/prospector/redis/prospector.go

```diff
@@ -0,0 +1,76 @@
+package redis
+
+import (
+	"github.com/elastic/beats/filebeat/channel"
+	"github.com/elastic/beats/filebeat/input/file"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+
+	"github.com/elastic/beats/filebeat/harvester"
+	rd "github.com/garyburd/redigo/redis"
+)
+
+// Prospector is a prospector for redis
+type Prospector struct {
+	started bool
+	outlet  channel.Outleter
+	config  config
+}
+
+// NewProspector creates a new redis prospector
+func NewProspector(cfg *common.Config, outlet channel.Outleter) (*Prospector, error) {
+
+	config := defaultConfig
+
+	err := cfg.Unpack(&config)
+	if err != nil {
+		return nil, err
+	}
+
+	p := &Prospector{
+		started: false,
+		outlet:  outlet,
+		config:  config,
+	}
+
+	return p, nil
+}
+
+// LoadStates loads the states
+func (p *Prospector) LoadStates(states []file.State) error {
+	return nil
+}
+
+// Run runs the prospector
+func (p *Prospector) Run() {
+
+	for _, host := range p.config.Hosts {
+		// TODO: Add timeout to connection -> configurable?
+		// TODO: should connections be reused across fetches?
+		c, err := rd.Dial("tcp", host)
+		if err != nil {
+			logp.Err("Hello world: %s", err)
+			return
+		}
+		defer c.Close()
+
+		logp.Info("Redis Prospector run !!!")
+
+		h := NewHarvester(p.outlet, c)
+		h.forwarder = harvester.Forwarder{}
+
+		// Start harvester for each host configure
+		err = h.Start()
+		if err != nil {
+			logp.Err("Error harvesting: %s", err)
+		}
+	}
+}
+
+func (p *Prospector) Stop() {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported method Prospector.Stop should have comment or be unexported